webpack/assets/javascripts/react_app/components/FactCharts/slice.js

```diff
@@ -0,0 +1,22 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  modalToDisplay: {},
+};
+
+const factChartSlice = createSlice({
+  name: 'factChart',
+  initialState,
+  reducers: {
+    openModal(state, { payload }) {
+      state.title = payload.title;
+      state.modalToDisplay = { [payload.id]: true };
+    },
+    closeModal(state) {
+      state.modalToDisplay = {};
+    },
+  },
+});
+
+export const { openModal, closeModal } = factChartSlice.actions;
+export default factChartSlice.reducer;
```

Comment:
can you add here the `selectFactChart` selector?
I believe this file is the right context for it, e.g:
```js
export const selectFactChart = state => state[slice.name];
```

Additionally, since the other selectors are using the API middleware, we can dispatch them in the `index.js` file and remove the `selector.js` file